code/items/weapons/TTTWeapon.cs

```diff
@@ -161,6 +161,24 @@ public override void Simulate(Client owner)
                 }
             }
 
+            //No idea where else to put this. So it goes here.
+            if (Input.Released(InputButton.View) && AmmoClip > 0) //Todo: Move this to be a bindable key (default Z???)
+            {
+                if (IsServer)
+                {
+                    var ammoBox = new BuckshotAmmo()
+                    {
+                        AmmoAmount = AmmoClip,
```

Comment:
`AmmoAmount = AmmoClip` doesn't seem to actually work, thats why there is a call to `.SetCurrentAmmo` on line 176, which changes the private variable and sets it appropriately.

I think this is likely due to my minimal understanding of how the fuck S&box operates it's client-server communication. Would definitely like solutions or changes in cleaning up the networking logic as I get used to it myself.

Reply:
To clarify. Variable change on 171 doesn't work. Function call on 176 does.

Reply:
@Alf21 probably has a better idea than I

Reply:
Is it because `AmmoAmmount` isn't a networked variable?

Reply:
Unsure. AmmoAmount cannot be set with a `Net` attribute since it's a virtual property.

Reply:
There might be a way to get around that with a different inheritance setup though.

Reply:
Let me see if I can come up with something